### PR TITLE
Pytdxmeasure: Add function for RTMR write

### PR DESCRIPTION
**Pull request by Ruoyu-y**
* Add extra function extend_rtmr to perform rtmr write

### CODE REVIEW

**Comment by Ruoyu-y** [comment]
@kenplusplus Here adds the function for rtmr extend. Please check. Rest of the test cases will be committed to stack-test as another PR.

**Review by kenplusplus — COMMENTED** [review]
Sound like we need design an actor like https://github.com/intel/tdx-tools/blob/7d80572f1952c23f4c55ab2dea8e5b0231b5d197/attestation/pytdxmeasure/pytdxmeasure/actor.py#L22 to allow user to run it via command line

File: attestation/pytdxmeasure/pytdxmeasure/rtmr.py [hunk 1]
```diff
@@ -22,3 +29,48 @@ def __eq__(self, other):
         bytearray_2, _ = other.get_bytes(0, RTMR.RTMR_LENGTH_BY_BYTES)
 
         return bytearray(bytearray_1) == bytearray(bytearray_2)
+
+    @staticmethod
+    def extend_rtmr(raw_extend_data, extend_rtmr_register):
```
Comment (kenplusplus) [inline attestation/pytdxmeasure/pytdxmeasure/rtmr.py:34]: Let user to provide the index of rtmr, instead of hard code 3.

**Review by kenplusplus — CHANGES_REQUESTED** [review]

**Comment by kenplusplus** [comment]
could you please rebase your branch with latest main, since https://github.com/intel/tdx-tools/commit/63c33b593b8669003667af9d7e8d755b8ff6378c will help to fix the issue of github action.

File: attestation/pytdxmeasure/pytdxmeasure/rtmr.py (hunk 1 above)
Comment (Ruoyu-y) [reply attestation/pytdxmeasure/pytdxmeasure/rtmr.py:34]: > Sound like we need design an actor like
> 
> https://github.com/intel/tdx-tools/blob/7d80572f1952c23f4c55ab2dea8e5b0231b5d197/attestation/pytdxmeasure/pytdxmeasure/actor.py#L22
> 
> to allow user to run it via command line

Sure. I will do that

Comment (Ruoyu-y) [reply attestation/pytdxmeasure/pytdxmeasure/rtmr.py:34]: I think currently the variable 'extend_rtmr_register' will give options to user on which index they would like to use. Let me change the variable name to 'extend_rtmr_index' to clarify that.

**Comment by Ruoyu-y** [comment]
> Sound like we need design an actor like
> 
> https://github.com/intel/tdx-tools/blob/7d80572f1952c23f4c55ab2dea8e5b0231b5d197/attestation/pytdxmeasure/pytdxmeasure/actor.py#L22
> 
> to allow user to run it via command line

Sure.

**Review by kenplusplus — APPROVED** [review]
LGTM

**Comment by kenplusplus** [comment]
https://github.com/intel/tdx-tools/actions/runs/4343937104/jobs/7589452040#step:6:101 pylint issue

File: attestation/pytdxmeasure/pytdxmeasure/rtmr.py (hunk 1 above)
Comment (kenplusplus) [reply attestation/pytdxmeasure/pytdxmeasure/rtmr.py]: ok.. my bad, not realize it...

**Review by kenplusplus — APPROVED** [review]
LGTM

**Pull request merged**